pkg/analyzer/image.go

```diff
@@ -387,3 +404,51 @@ func isWEBPAnimated(file io.Reader) (bool, error) {
 	}
 	return frames > 1, nil
 }
+
+// MatchImage compares the link with the given hashes to determine if they match.
+func MatchImage(link manifest.Link, hashes manifest.HashList) (bool, error) {
+	if link.MediaType == nil || !link.MediaType.IsBitmap() {
+		return false, errors.New("link is not to an image that can be matched")
+	}
+
+	linkHashes := link.Properties.Hash()
+	if len(linkHashes) == 0 {
+		// No hashes in the link, we can't match it
+		return false, nil
+	}
+	for _, hash := range hashes {
+		if v, ok := linkHashes.Find(hash.Algorithm); ok {
+			if v.Equal(hash) {
+				// Simple equality
+				return true, nil
+			}
+
+			// Special distance-based matching for perceptual hashes
+			if v.Algorithm == manifest.HashAlgorithmPhashDCT {
```

Comment:
This is perfectly fine for this PR. One possible improvement to support extensibility of hash algorithms would be to delegate the comparison to a set of `HashMatcher`, with the default one doing a basic equality for example.

Reply:
This might be something to add as an issue. I definitely agree if this gets any more complicated we'll need that